pe.c

```diff
@@ -1074,18 +1075,23 @@ handle_image (void *data, unsigned int datasize,
 				       entry->vendor_version,
 				       entry->vendor_url);
 			}
+			res = parse_sbat_var(&sbat_var_entries);
+			if (res < 0)
+				console_print(L"could not parse SBAT variable");
 		} else {
 			perror(L"SBAT data not found\n");
 			return EFI_UNSUPPORTED;
 		}
-
 		efi_status = verify_buffer(data, datasize,
 					   &context, sha256hash, sha1hash);
-
+		if (sbat.entries)
+			efi_status = verify_sbat(&sbat, &sbat_var_entries);
+			/* Add EFI_SECURITY_VIOLATION if variable is set,
+			and binary has no SBAT section ?*/
 		if (sbat.entries)
```

Comment:
Drop this and make it all part of the same block if `sbat.entries` is set. This condition should always be true anyways.